### ISSUE TITLE
Chat history not deleted when deleting a room

### ISSUE THREAD

**Issue by pkukkon** [igniterealtime/openfire-monitoring-plugin#135]
If a room gets deleted in Openfire, it is deleted from the database as well (`mucroom`) but tables `ofmucconversationlog` and via that, `ofmessagearchive`, still have references to `roomID` that no longer exists. While it is debatable whether deleting a room should also delete all messages of that room, that kind of sounds like the expected behavior?

**Comment by akrherz** [comment]
Yeah, this gets into tricky business logic about what should be done.  Deleting logs would be bad as the point of the plugin is often for compliance.  I suspect we need to have some sort of flag on `ofmucroom` table to denote 'archived' rooms, such that a given room `name` could have duplicates to account for a room that gets deleted and then is created again with the same name.
